sqlitecluster/SQLiteClusterMessenger.cpp

```diff
@@ -11,9 +11,12 @@ bool SQLiteClusterMessenger::sendToLeader(BedrockCommand& command) {
     string leaderAddress;
     auto _nodeCopy = atomic_load(&_node);
     if (_nodeCopy) {
+        // peerList is const, so we can safely read from it in  multiple threads without locking, similarly,
+        // peer->commandAddress is atomic.
         for (SQLiteNode::Peer* peer : _nodeCopy->peerList) {
-            if (peer->state == STCPNode::LEADING && !peer->commandAddress.load().empty()) {
-                leaderAddress = peer->commandAddress;
+            string peerCommandAddress = peer->commandAddress;
+            if (peer->state == STCPNode::LEADING && !peerCommandAddress.empty()) {
+                leaderAddress = peerCommandAddress;
```

Comment:
Can you explain this change? From the comment you added, it sounds like assigning `peer->commandAddress` to `peerCommandAddress` is not necessary... unless `load()` does something I wouldn't expect? NAB though since it does not seem like there's any harm to this change.

Reply:
These two lines do the same thing:
```
string peerCommandAddress = peer->commandAddress;
string peerCommandAddress = peer->commandAddress.load();
```

One calls `load()` explicitly, and the other ends up calling the cast operator, which you can see has the exact same implementation as `load()`:
https://github.com/Expensify/Bedrock/blob/main/libstuff/libstuff.h#L306-L313

So whether you call `load()` or not in this case doesn't make any difference.

What this change does is check and assign a value that is guaranteed to be the same. Because we now assign to a temporary varaible (`peerCommandAddress`), and then check whether that is empty, and if not, assign it to `leaderAddress`, we know that the thing we checked for emptiness is also the thing we assigned.

In the old code, we do two separate accesses to `peerCommandAddress`, one to check for emptiness, and one for the assignment, and there's no guarantee another thread hasn't changed the value in between those checks.

Reply:
Ah good call about the race condition, thanks for explaining